src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/MarkPaymentPaidStateHandler.java

```diff
@@ -0,0 +1,78 @@
+package uk.gov.hmcts.reform.iacaseapi.domain.handlers.presubmit;
+
+import static java.util.Objects.requireNonNull;
+import static uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCaseFieldDefinition.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AppealType;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.AsylumCase;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.State;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.PaymentStatus;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackStateHandler;
+
+@Component
+public class MarkPaymentPaidStateHandler implements PreSubmitCallbackStateHandler<AsylumCase> {
+
+    private final boolean isfeePaymentEnabled;
+
+    public MarkPaymentPaidStateHandler(
+        @Value("${featureFlag.isfeePaymentEnabled}") boolean isfeePaymentEnabled
+    ) {
+        this.isfeePaymentEnabled = isfeePaymentEnabled;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && callback.getEvent() == Event.MARK_APPEAL_PAID
+               && isfeePaymentEnabled;
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback,
+        PreSubmitCallbackResponse<AsylumCase> callbackResponse
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        final State currentState =
+            callback
+                .getCaseDetails()
+                .getState();
+
+        asylumCase.write(PAYMENT_STATUS, PaymentStatus.PAID);
+
+        String paymentDate = asylumCase.read(PAID_DATE, String.class)
+            .orElseThrow(() -> new IllegalStateException("Paid date is not present"));
+        asylumCase.write(PAYMENT_DATE,
```

Comment:
Reading and writing PAID_DATE with format. Is this required if its written in the right format ?

Reply:
When writing to different place holder the format is lost, so I'm formatted it before writing